### ISSUE TITLE
Validity is not properly checking rules proceeding closing curly brace

### ISSUE THREAD

**Issue by awayken** [rossPatton/stylint#63]
If a line of Stylus code ends with a curly brace, the code is not properly checked for validity. I was able to specify made up properties, use invalid values and even just type text that doesn't follow CSS rules at all.

invalid.styl
```stylus
// Passes but shouldn't
body { dog: irish-setter; banana: brown; hair-length: 10em; Lorem ipsum }
body {
    dog: irish-setter; banana: brown; hair-length: 10em; Lorem ipsum }

// Fails and should
body {
    dog: irish-setter; banana: brown; hair-length: 10em; Lorem ipsum
}
body {
    dog: irish-setter;
    banana: brown;
    hair-length: 10em;
    Lorem ipsum
}
```

invalid.json
```json
{
    "alphabetical": true,
    "borderNone": true,
    "brackets": false,
    "colons": false,
    "colors": false,
    "commaSpace": true,
    "commentSpace": false,
    "cssLiteral": false,
    "depthLimit": false,
    "duplicates": true,
    "efficient": true,
    "enforceBlockStyle": false,
    "enforceVarStyle": false,
    "extendPref": false,
    "globalDupe": false,
    "indentSpaces": 4,
    "leadingZero": true,
    "maxWarnings": 10,
    "maxWarningsKill": false,
    "mixed": false,
    "namingConvention": false,
    "parenSpace": false,
    "placeholders": true,
    "semicolons": false,
    "universal": true,
    "valid": true,
    "whitespace": true,
    "zeroUnits": true,
    "zIndexDuplicates": false,
    "zIndexNormalize": false
}
```

Output:
```bash
Warning:  property is not valid
File: invalid.styl
Line: 8: dog: irish-setter; banana: brown; hair-length: 10em; Lorem ipsum

Warning:  property is not valid
File: invalid.styl
Line: 11: dog: irish-setter;

Warning:  property is not valid
File: invalid.styl
Line: 12: banana: brown;

Warning:  property is not valid
File: invalid.styl
Line: 13: hair-length: 10em;

Warning:  property is not valid
File: invalid.styl
Line: 14: Lorem ipsum


5 Warnings
```

I think an interesting testament to how crucial this project is when you look at how Stylus itself treats this same file. It doesn't care at all. It goes right ahead and changes the file into what it thinks is CSS.

invalid.css
```css
body {
  dog: irish-setter;
  banana: #a52a2a;
  hair-length: 10em;
  Lorem: ipsum;
}
body {
  dog: irish-setter;
  banana: #a52a2a;
  hair-length: 10em;
  Lorem: ipsum;
}
body {
  dog: irish-setter;
  banana: #a52a2a;
  hair-length: 10em;
  Lorem: ipsum;
}
body {
  dog: irish-setter;
  banana: #a52a2a;
  hair-length: 10em;
  Lorem: ipsum;
}
```

It's never been more clear to me how important and useful Stylint is for maintaining and writing sane, valid and professional Stylus code.

**Comment by rossPatton** [comment]
So, i'm able to fix it so this example works;
```
body {
    dog: irish-setter; banana: brown; hair-length: 10em; Lorem ipsum }
```

but i think it will take more work for this one:
```
body { dog: irish-setter; banana: brown; hair-length: 10em; Lorem ipsum }
```

Basically, valid just checks properties, not values. It sees that body is a correct value, and moves on.

I think to solve this i'd need to some checking to if a rule is a one-liner (which is bad practice imo) and if so, parse it into something that be checked against (ie, separate the properties and values, and run the valid check against the properties).

I think that would be a slightly bigger lift, and i'd probably want to add another option as a result of it, to discourage one-liners. 

Something like `oneLiners: false`, which would determine if a rule contains more than one property:value pair, and then throw a warning.

**Comment by rossPatton** [comment]
i think the best next step would be to close this issue out, since i've made fixes based on this that willl go up soon, and create a new one around doing this one-liner check

**Comment by awayken** [comment]
The only way for a one-liner to be possible in Stylus is with a curly brace. Otherwise Stylus throws a big, nasty error. Does being able to make that assumption help with checking the single line instance?

In any case, I agree with the new configuration. I'll create an issue proposing we add that as an option.

**Comment by rossPatton** [comment]
Looking to see if `{` isn't last might make testing it possible, i'll give it a go

**Comment by rossPatton** [comment]
i feel like once https://github.com/rossPatton/stylint/issues/77 is live this issue can be considered resolved.
